google/cloud/spanner_dbapi/cursor.py

```diff
@@ -269,9 +259,18 @@ def _execute(self, sql, args=None, call_from_execute_many=False):
             ):
                 self._handle_DQL(sql, args or None)
             elif self._parsed_statement.statement_type == StatementType.DDL:
-                self._batch_DDLs(sql)
-                if not self.connection._client_transaction_started:
-                    self.connection.run_prior_DDL_statements()
+                if not self.connection.buffer_ddl_statements:
+                    if not self.connection._client_transaction_started:
+                        self._batch_DDLs(sql)
+                        self.connection.run_prior_DDL_statements()
+                    else:
+                        raise ProgrammingError(
```

Comment:
Isn't this also a change in behavior? It's a bit hard to read with all the changes here, but I think that previously we would silently accept this, and now we are returning an error?

Reply:
No this code path would only be reached when `buffer_ddl_statements` flag is disabled which was not the case earlier

Reply:
ack